integration/utmp_integration_test.go

```diff
@@ -0,0 +1,254 @@
+//+build test_as_root
+
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/bpf"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/pam"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/regular"
+	"github.com/gravitational/teleport/lib/srv/uacc"
+	"github.com/gravitational/teleport/lib/sshutils"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+	. "gopkg.in/check.v1"
+)
+
+// teleportTestUser is additional user used for tests
+const teleportTestUser = "teleport-test"
+
+// wildcardAllow is used in tests to allow access to all labels.
+var wildcardAllow = services.Labels{
+	services.Wildcard: []string{services.Wildcard},
+}
+
+type SrvCtx struct {
+	srv        *regular.Server
+	signer     ssh.Signer
+	server     *auth.TestTLSServer
+	testServer *auth.TestAuthServer
+	clock      clockwork.FakeClock
+	nodeClient *auth.Client
+	nodeID     string
+}
+
+// TestUTMP tests that user accounting is done on supported systems.
+func TestUTMPEntryExists(t *testing.T) {
+	s := &SrvCtx{}
+	s.SetUpContext(t)
```

Comment:
You might need to clean up (e.g. stop servers) as well to make the tests re-entrancy friendly.